lib/linguist/heuristics.rb

```diff
@@ -17,9 +17,8 @@ def self.call(blob, candidates)
       data = blob.data
 
       @heuristics.each do |heuristic|
-        if heuristic.matches?(blob.name)
-          languages = Array(heuristic.call(data))
-          return languages if languages.any? || languages.all? { |l| candidates.include?(l) }
```

Comment:
Oo :scream: You're right, that should've been an AND. And I'm the one who switched it to an OR :sweat: 

This was for cases where we'd use the same set of heuristic rules for two extensions with different candidates. For example if we use heuristic rules on C, C++, and Objective-C for both `.h` (the three languages are candidates) and `.inc` (only C and C++ are candidates). In that case, the extra check would ensure Objective-C is never returned for the `.inc` extension.

We currently don't have any such case, so we're probably fine.